keps/487-kubectl-plugin/README.md

```diff
@@ -0,0 +1,545 @@
+motivation section can optionally provide links to [experience reports] to
+demonstrate the interest in a KEP within the wider Kubernetes community.
+
+[experience reports]: https://github.com/golang/go/wiki/ExperienceReports
+-->
+
+### Goals
+
+A successful plugin should be able to answer the following questions:
+
+1. What jobs are there in the user-queue? (`jobs/describe`)
+1. What jobs are there in this specific namespace? (`jobs/describe`)
+2. What jobs are there in this specific state? (`jobs/describe`)
+3. Why is my workload pending? (`describe`)
+4. Was there an error admitting my workload? (`describe`) 
+5. Is a ClusterQueue misconfigured or some other issue? (this is more of an admin command?) (`queues` or `describe-queues`)
+6. All of the above, but instead of a table I want json/yaml (add `-o yaml/json`)
+
+
+<!--
+List the specific goals of the KEP. What is it trying to achieve? How will we
+know that this has succeeded?
+-->
+
+### Non-Goals
+
+- Serve as a duplicate implementation of information available via kubectl already
+
+<!--
+What is out of scope for this KEP? Listing non-goals helps to focus discussion
+and make progress.
+-->
+
+## Proposal
+
+We propose creating a command-line tool that can serve as a Kubectl plugin that exposes this missing information. We also propose using a design strategy that mimics existing tools that are available for other workload managers across HPC and cloud that users are comforable with to ease adoption of both the tool and approach of submitting jobs to Kubernetes. The main command-line interactions will take the following shape:
+
+```bash
+# As a Kubectl plugin
+kubectl kueue <options> <command< <args>
+
+# As a standalone command-line tool
+kuctl <options> <command> <args>
+```
+
+For comparison, the Armada team developed a [general client](https://github.com/armadaproject/armada/tree/master/cmd/armadactl) that uses this strategy to be renamed and fold into kubectl. This is an ideal approach because it presents an interface to kueue to make it feel more like a standalone job manager (and not force folks to live in Kubernetes abstractions, at least entirely). 
+
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+### User Stories (Optional)
+
+<!--
+Detail the things that people will be able to do if this KEP is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+#### Story 1
+
+As a user submitting jobs, I want to easily see the status of an entire Workload,
```

Comment:
From what I understand of Kueue, Kueue has a concept of a workload which is internal and hidden from the user.  I think that in order to answer the question of if a "workload" is running/finished you would actually want the plugin to reach out to the actual workload.  

In the case of Job you can look at the status or something like that.  I don't know if a Kueue workload has these concepts.

Reply:
I'm not familiar enough either - I've only run very basic examples using Kueue with jobs.

Reply:
Ignore me!  https://github.com/kubernetes-sigs/kueue/blob/main/apis/kueue/v1beta1/workload_types.go#L130

Looks like it may be available as a condition.

Reply:
@kannon92 I will take many requests but one thing I will not do is ignore you! :)